very epic temp ban pull request + fix + added myself to the about command hehe

iakzs opened the pull request:
im too cool fr

iakzs committed:
ok bro

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
temp ban

iakzs committed:
maybe a fix if an interaction error shows up

iakzs committed:
added expired bans

iakzs committed:
added the expired ban check at the start

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
added myself to the about hehe

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
yes

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

Commit:
cool

iakzs commented:
i wonder why are there so many merge branch dev of .. into dev

Commit:
forgot the bun.lockb

Golem642 commented on src/commands/about.ts:
```diff
@@ -37,7 +37,7 @@ export default class About {
           value: [
             "**Founder**: Goos",
-            "**Developers**: Dimkauzh, Froxcey, Golem64, MQuery, Nikkerudon, Spectrum, ThatBOI",
+            "**Developers**: Dimkauzh, Froxcey, Golem64, Koslz, MQuery, Nikkerudon, Spectrum, ThatBOI",
```
Lol

Golem642 commented on src/utils/unbanScheduler.ts:
```diff
@@ -0,0 +1,58 @@
+  const maxDelay = 2147483647; 
+
+  if (delay > maxDelay) {
+    const remainingDelay = delay - maxDelay;
+    const timeout = setTimeout(() => {
+      scheduleUnban(client, guildId, userId, remainingDelay);
+    }, maxDelay);
```
What happens if it goes negative? I never tested that, will it count as a timer of 0 ?

iakzs replied:
I will check later. And if unbans instantly or something I will add a fix

Golem642 reviewed (commented):
It works, but it's definitely not the best way to do it. Having setTimeouts everywhere means having multiple processes that are waiting, and I'm not sure that's memory efficient. I'm gonna take a better look at it later and see what can be optimized

iakzs commented:
I can't change anything rn, when I get the time to do so I will

iakzs committed:
updat unban scheduler

setTimeout (i think) only uses some memory at time of making the ban and the time of unbanning. but if i (1st idea i had) made it check every 1 minute or something would use more memory. i might be wrong

iakzs commented:
> ...Having setTimeouts everywhere means having multiple processes that are waiting, and I'm not sure that's memory efficient...

Maybe, but from what I know the setTimeout uses only memory on the time of trigger (the ts and when the time is off). Please correct me if I'm wrong.

MrSerge01 reviewed (approved):
some stuff that you did here are already in my WIP commit but ill work it out frfr :pensive:
LGTM,
\- The Goos.
